R/g.part5.definedays.R

```diff
@@ -5,10 +5,21 @@ g.part5.definedays = function(nightsi, wi, indjump, nightsi_bu,
   Nts = nrow(ts)
   lastDay = FALSE
   # define local functions ----
-  qwindow2timestamp = function(qwindow) {
+  qwindow2timestamp = function(qwindow, epochSize) {
     H = floor(qwindow)
     M = floor((qwindow - H) * 60)
     S = floor((qwindow - H - M/60) * 60 * 60)
+    expected_S = seq(0, 60, by = epochSize)
+    if (any(!S %in% expected_S)) {
+      revise = which(!S %in% expected_S)
+      for (si in revise) {
+        S[si] = expected_S[which.min(abs(expected_S - S[si]))]
```

Comment:
Let's avoid single letter object names H M and S, and instead rename them to hour, minute, and second. single letter object names are very hard to search when trying to debug code. It is better to use slightly longer names that clarify what you are calculating.

Object `revise` is also not specific, maybe good to replace by seconds_tobe_revised or similar.